[15.0][MIG] excel_import_export, excel_import_export_demo: migration

Mantux11 opened the pull request:
Migration to version `15.0`. Changes by pre-commit and changes because of Odoo code changes.

kittiu committed:
Excel Import/Export/Report  (#1522)

* [ADD] v12 excel_import_export

* Change from eval() to safe_evel()

* Change variable to format to style, as fomat is a common python function

:100644 100644 00ee3d9f... e9e48d87... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 01e5b9f5... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 17d3964d... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 933ce0dc... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 3b1217e8 M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 a7d6adc5 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 e3826e08 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 c8481487 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 ed8c9fc7 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 69aa6ea0 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 c9733b95 A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 bb3ea32e A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

kittiu committed:
Excel Import/Export/Report  (#1522)

* [ADD] v12 excel_import_export

* Change from eval() to safe_evel()

* Change variable to format to style, as fomat is a common python function

:100644 100644 00ee3d9f... e9e48d87... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 01e5b9f5... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 17d3964d... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 933ce0dc... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 3b1217e8 M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 a7d6adc5 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 e3826e08 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 c8481487 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 ed8c9fc7 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 69aa6ea0 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 c9733b95 A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 bb3ea32e A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-travis committed:
[UPD] Update excel_import_export_demo.pot

kittiu committed:
[12.0][FIX] excel_import_export

wrong use of dict.iteritems() for python3

kittiu committed:
[12.0][FIX] excel_import_export

wrong use of dict.iteritems() for python3

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] icon.png

kittiu committed:
[12.0][FIX] excel_import_export

Fix Import Excel wizard, cannot download template as in v12
core Odoo prevent download default binary.

kittiu committed:
[12.0][FIX] excel_import_export

Fix Import Excel wizard, cannot download template as in v12
core Odoo prevent download default binary.

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][FIX] excel_import_export, use sudo() to get param

kittiu committed:
[12.0][IMP] excel_import_export, excel_import_export_demo

Add report action feature and new examples

kittiu committed:
[12.0][IMP] excel_import_export, excel_import_export_demo

Add report action feature and new examples

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-travis committed:
[UPD] Update excel_import_export_demo.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][FIX] excel_import_export, fix wrong file name on report_action

When mark to convert xlsx to csv, csv file name is not used

Saran440 committed:
[12.0][FIX] excel_import_export - Style sum footer to number

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 2.9% (5 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 24.1% (41 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (170 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (170 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 71.9% (23 of 32 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export_demo
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export_demo/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (32 of 32 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export_demo
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export_demo/zh_CN/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][ENH] excel_import_export, faster export

Previously export is slow on large number of rows.
This fix ensure that sheet.insert_rows is called only once for all rows,
instead of insert 1 row for every rows

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

oca-travis committed:
[UPD] Update excel_import_export_demo.pot

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export_demo
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export_demo/

kittiu committed:
[12.0][FIX] excel_import_export, problem on import with datetime field

Can't import if the import file has date time field

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

kittiu committed:
[IMP] : black, isort, prettier

kittiu committed:
[IMP] : black, isort, prettier

kittiu committed:
[13.0][MIG] excel_import_export, excel_import_export_demo

kittiu committed:
[13.0][MIG] excel_import_export, excel_import_export_demo

kittiu committed:
[IMP] excel_import_export: black, isort, prettier

kittiu committed:
[14.0][MIG] excel_import_export, excel_import_export_demo

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-travis committed:
[UPD] Update excel_import_export_demo.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-travis committed:
[UPD] Update excel_import_export_demo.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export_demo
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export_demo/

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-travis committed:
[UPD] Update excel_import_export_demo.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export_demo
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export_demo/

alessandro-fiorino committed:
Added translation using Weblate (Italian)

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 5.5% (11 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

ofonseca-pyming committed:
Added translation using Weblate (Catalan)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

OCA-git-bot committed:
[IMP] update dotfiles

Mantux11 commented:
@kittiu Hey. Because of your migration to v13 and v14, I want some help from you. I think, you know how JS worked here and mostly all functionality what JS code does here. Personally, I don't know probably anything about JS, and I saw that Odoo deleted `ActionManager` and I suggest that everything was moved to services or somewhere else. So my questions are:

1. How JS code worked here, what was functionality of JS code, why it was used here?
2. Maybe you could suggest, where I could try to look to understand what is going on here? Maybe you know where everything was moved, what was changed and where I could insert these functions? Some files to look, classes and etc. 

Thank you! :wink:

kittiu commented:
@Mantux11 you are wrong that I know what I was doing about JS :laughing: 
I just try to imitate the work from other module, i..e, report_csv, report_xlsx.
Now, I saw this already migrated to v15 https://github.com/OCA/reporting-engine/blob/15.0/report_xlsx/static/src/js/report/action_manager_report.esm.js

:)

Mantux11 commented:
@kittiu Thanks for your answer. I will try to change what it's needed to be changed. :D

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

kittiu commented:
Not sure this is ready for review. Functional test couple of action without success.

1. Sales > Reporting > Partner List Report
![image](https://user-images.githubusercontent.com/1973598/168535166-7f4e3ef0-d8a7-47a5-a9d1-9f390165584e.png)
2. Sales > Sales Order > Quotation / Order (.xlsx) ----> not in the print action yet.
3. Technical > Excel Import/Export > Sample Lead Report, error when click to export.
![image](https://user-images.githubusercontent.com/1973598/168535489-55dfd83e-f1a3-4c26-b617-fba733465098.png)

Mantux11 commented:
> Not sure this is ready for review. Functional test couple of action without success.
> 
> 
> 
> 1. Sales > Reporting > Partner List Report
> 
> ![image](https://user-images.githubusercontent.com/1973598/168535166-7f4e3ef0-d8a7-47a5-a9d1-9f390165584e.png)
> 
> 2. Sales > Sales Order > Quotation / Order (.xlsx) ----> not in the print action yet.
> 
> 3. Technical > Excel Import/Export > Sample Lead Report, error when click to export.
> 
> ![image](https://user-images.githubusercontent.com/1973598/168535489-55dfd83e-f1a3-4c26-b617-fba733465098.png)
> 
> 
> 
> 



Okey, I will test more on that. Thanks, for review

Mantux11 commented:
@kittiu Talking about Sample Lead Report, you also missed that, this isn't working in 14 version, because `planned_revenue` field is deleted from 14 version. So I changed to `expected_revenue`, why I changed to that, you could see comparing 13 and 14 `crm.lead` core code. Partner List report is fixed. Talking about SO report, I see that when you press Export XLSX, you could choose what template to use in wizard. Maybe I'm wrong, but this should work as it is? Or there should be one more report? It would be really appreciated, If you could give me more example's on SO report, how it should look like,.

luisg123v committed:
[REF] excel_import_export: remove xlrd & xlwt from external dependencies

Because those dependencies are already part of Odoo's base dependencies.

OCA-git-bot committed:
excel_import_export 14.0.1.1.0

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (200 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/es_AR/

gaikaz commented:
@OCA/server-environment-maintainers 
Sorry to be pinging, but this is a valid migration that is just not getting any attention for 3 months now. Wouldn't want to be considered stale and closed.
Could we please get a maintainer review or two? 🙏🏻

kittiu commented:
@gaikaz have you tested this? Can you approve this? Thanks!

kittiu commented:
Ping @Saran440 @ps-tubtim

gaikaz commented:
@kittiu 
We'll be migrating one of our modules that depend on this, hopefully over next week. 
Will do my review too.

Du-ma commented on excel_import_export/static/src/js/report/action_manager_report.esm.js:
```diff
@@ -0,0 +1,56 @@
+                const context = encodeURIComponent(JSON.stringify(actionContext));
+                url += `?options=${action_options}&context=${context}`;
+            } else {
+                if (actionContext.active_ids) {
+                    url += `/${actionContext.active_ids.join(",")}`;
+                }
+                if (type === "excel") {
+                    const context = encodeURIComponent(
+                        JSON.stringify(env.services.user.context)
+                    );
+                    url += `?context=${context}`;
+                }
+            }
+            env.services.ui.block();
+            try {
+                await download({
+                    url: "/report/download",
+                    data: {
+                        data: JSON.stringify([url, action.report_type]),
+                        context: JSON.stringify(env.services.user.context),
+                    },
+                });
+            } finally {
+                console.log("pim");
+                env.services.ui.unblock();
+            }
+            console.log("pams");
+            const onClose = options.onClose;
+            if (action.close_on_report_download) {
+                return env.services.action.doAction(
+                    {type: "ir.actions.act_window_close"},
+                    {onClose}
+                );
+            } else if (onClose) {
+                onClose();
+            }
+            return Promise.resolve(true);
+        }
+        return Promise.resolve(false);
+    });
```
```suggestion
function getReportUrl({report_name, context, data}, env) {
    // Rough copy of action_service.js _getReportUrl method.
    let url = `/report/excel/${report_name}`;
    const actionContext = context || {};
    if (data && JSON.stringify(data) !== "{}") {
        const encodedOptions = encodeURIComponent(JSON.stringify(data));
        const encodedContext = encodeURIComponent(JSON.stringify(actionContext));
        return `${url}?options=${encodedOptions}&context=${encodedContext}`;
    }
    if (actionContext.active_ids) {
        url += `/${actionContext.active_ids.join(",")}`;
    }
    const userContext = encodeURIComponent(JSON.stringify(env.services.user.context));
    return `${url}?context=${userContext}`;
}

async function triggerDownload(action, {onClose}, env) {
    // Rough copy of action_service.js _triggerDownload method.
    env.services.ui.block();
    try {
        await download({
            url: "/report/download",
            data: {
                data: JSON.stringify([getReportUrl(action, env), "excel"]),
                context: JSON.stringify(env.services.user.context),
            },
        });
    } finally {
        env.services.ui.unblock();
    }
    if (action.close_on_report_download) {
        return env.services.action.doAction(
            {type: "ir.actions.act_window_close"},
            {onClose}
        )
    }
    if (onClose) {
        onClose();
    }
}

registry
    .category("ir.actions.report handlers")
    .add("xlsx_handler", async function (action, options, env) {
        if (action.report_type === "excel") {
            await triggerDownload(action, options, env);
            return true;
        }
        return false;
    });
```
How about something like this?
Removed the `console.log`s as well as made the code easier to read (in my opinion).
Hard-coded in some elements specific to this report, for example - How would we have a different type of report rather than `excel`, when it's in the if block?

Mantux11 replied:
Thanks @Du-ma. I will test that ASAP.

Du-ma replied:
Whoops, forgot something, edited my suggestion

Saran440 reviewed (approved):
Test with demo :+1:

GreyShim commented:
Hello guys,

Is it possible to merge the pull request ? I need this module for a project

Regards

kittiu commented:
@Mantux11 looks like pre-commit is failing. Can you check please?

GreyShim commented:
@Mantux11  

do not forget to integrate this change : https://github.com/OCA/server-tools/pull/2283/files
it's very important, data are deleted without this

Mantux11 commented:
@GreyShim I would wait for that PR to be merged. And I would include that here.

Mantux11 commented:
@kittiu, @GreyShim, @Du-ma Could you review it now, please? This module is waiting in the line a long time ago. I changed what @GreyShim said that, and also, included what @luc-demeyer suggested in that PR above.

norlinhenrik reviewed (approved):
Thank you for migrating and fixing this module!

models/common.py
`line = "{}.{}".format("xls", uuid.uuid4()) + "," + line`
To be consistent, please change to
`line = "{}.{}".format("--excel_import_export--", uuid.uuid4()) + "," + line`
with underscore instead of hyphen. (I cannot write underscore since it is evaluated as bold.)

gaikaz commented:
@Mantux11 Henrik here has very good suggestion. 

@norlinhenrik 
You can do `\_` and have \_\_as many underscores as you want\_\_😉

Du-ma commented:
Pls don't copy paste that

Mantux11 commented:
@norlinhenrik Changed that. @Du-ma Yes I know, how you like to use `f"{}"` string :D

Du-ma commented on excel_import_export/models/ir_report.py:
```diff
@@ -41,4 +39,4 @@ def _get_report_from_name(self, report_name):
         ]
         context = self.env["res.users"].context_get()
-        return report_obj.with_context(context).search(conditions, limit=1)
+        return report_obj.with_context(context=context).search(conditions, limit=1)
```
```suggestion
        return report_obj.with_context(**context).search(conditions, limit=1)
```
Why do we even need this `context`???
Don't we already have one?
Are we missing something?

norlinhenrik replied:
I think the context may be removed.

Du-ma commented on excel_import_export/wizard/import_xlsx_wizard.py:
```diff
@@ -1,9 +1,13 @@
 from odoo.exceptions import RedirectWarning, ValidationError
 
+_logger = logging.getLogger(__name__)
+
```
```suggestion
```

Du-ma commented on excel_import_export/wizard/import_xlsx_wizard.py:
```diff
@@ -1,9 +1,13 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
 
+import logging
+
```
```suggestion
```

Du-ma commented on excel_import_export/wizard/import_xlsx_wizard.py:
```diff
@@ -98,8 +102,8 @@ def default_get(self, fields):
-                    _('File "%s" not found in template, %s.')
-                    % (template.fname, template.name),
+                    _('File "%(fname)s" not found in template, %(name)s.')
+                    % {"fname": template.fname, "name": template.name},
```
```suggestion
                    _('File "%(fname)s" not found in template, %(name)s.', fname=template.fname, name=template.name),
```
For future reference.

Du-ma commented:
Also a general question, why is this module not in `oca/reporting-engine`?

AungKoKoLin1997 commented on excel_import_export/__manifest__.py:
```diff
@@ -0,0 +1,30 @@
+    "website": "https://github.com/OCA/server-tools",
+    "category": "Tools",
+    "depends": ["mail"],
+    "external_dependencies": {"python": ["xlrd", "xlwt", "openpyxl"]},
```
```suggestion
    "external_dependencies": {"python": ["openpyxl"]},
```

AungKoKoLin1997 reviewed (commented):
Test with demo data: work as expected
For the functional, there is a minor things.

We can remove `xlrd` and `xlwt` from external dependencies because these two are part of base odoo requirements.

Sales > Sales Order > Quotation / Order (.xlsx) ----> not in the print action yet.
We need to click manually to add this report under print action in report definition. But I think this is not a problem. We can export under action.

![Screenshot from 2022-12-14 17-32-08](https://user-images.githubusercontent.com/45355704/207578962-4c41a8b4-3962-46b8-94c9-f8087cb1a07c.png)

norlinhenrik commented:
> Also a general question, why is this module not in `oca/reporting-engine`?

I use this module to import data. I prefer keeping it in `oca/server-tools`.

kittiu commented:
> Also a general question, why is this module not in `oca/reporting-engine`?

This was discussed before, for the fact that it can import too.l, and many time used as data import/export. So being just report does not comply. It is more like tools to build things.

Mantux11 commented:
@Du-ma Included your suggestions. @AungKoKoLin1997 Also deleted dependencies. And about action, so if it's not a problem, I have to add somewhere this action, or leave it as it is?

AungKoKoLin1997 commented on excel_import_export_demo/report_action/sale_order/report.xml:
```diff
@@ -0,0 +1,15 @@
+        <field name="model">sale.order</field>
+        <field name="report_type">excel</field>
+        <field name="report_name">sale_order_form.xlsx</field>
+        <field name="report_file">sale_order</field>
```
```suggestion
        <field name="report_file">sale_order</field>
        <field name="binding_model_id" ref="sale.model_sale_order"/>
        <field name="binding_type">report</field>
```

AungKoKoLin1997 replied:
Please add these lines for 
Sales > Sales Order > Quotation / Order (.xlsx) ----> not in the print action yet.

Mantux11 replied:
Added. Thanks :)

AungKoKoLin1997 commented:
> We can remove `xlrd` and `xlwt` from external dependencies because these two are part of base odoo requirements.

@Mantux11 Sorry. I didn't see this PR in 14.0. Can you please pay attention to this PR https://github.com/OCA/server-tools/pull/2349 and add in your commit history?

Mantux11 commented:
@AungKoKoLin1997 I included all commit's from `14.0`, Could you review, and maybe merge this?

Commit:
[IMP] excel_import_export, excel_import_export_demo: black, isort, prettier

Commit:
[15.0][MIG] excel_import_export, excel_import_export_demo: Migration

Mantux11 commented:
@AungKoKoLin1997 I included all newest commit's history from `14.0` with commit that removes mentioned libaries, Could you review, and maybe merge this?

Or @kittiu Could you merge?

Commit:
[15.0][FIX] excel_import_export, excel_import_export_demo:

fix from PR #2283 together with @luc-demeyer suggestion #2283 (review)

gaikaz commented:
@kittiu 
I can finally approve this 🎉 
We have a client with a big excel report that uses this module for generating it and after we now have migrated the client that uses this module while internally fixing any found issues with @Mantux11, it now definitely works.

kittiu commented:
Thank you all but I don't have power to merge. @dreispt can you help merge this? Thanks!

gaikaz commented:
@OCA/server-environment-maintainers 
Pretty please, review this 🙏🏻 This has a lot of approves with functional tests as well. 
If we (Via laurea) find any issues, we'll make a PR to fix. Pinky promise.

dreispt commented:
/ocabot merge nobump
